Drivers/Src/Display.c

```diff
@@ -1,69 +1,190 @@
 #include "bsp.h"
 #include "Display.h"
+#include <math.h>
+#include <string.h>
 
-#define GECKO_DATA_LENGTH 5*2
+#define NEXTION_INSTRUCTION_SUCCESSFUL 0x01ffffff
 
-/*
- * Encodes a byte into two hex digits and places them at buf
- * @param byte the given data byte
- * @param buf the output buffer
- * @return void
+static const char *DELIMITER = ".";
+static const char *ASSIGNMENT = "=";
+static const char *TERMINATOR = "\xff\xff\xff";
+static const char *NO_ERROR = "No Error";
+
+// Color defintions for display
+static const uint16_t NEXTION_GREEN = 2016;
+static const uint16_t NEXTION_RED = 63488;
+static const uint16_t NEXTION_LIGHT_GREY = 42260;
+static const uint16_t NEXTION_DARK_GREY = 23275;
+static const uint16_t NEXTION_BURNT_ORANGE = 51872;
+
+static inline int IsNextionFailure(uint32_t val) {
+    return ((val & ~0x00FFFFFF) != (1 << 24));
+}
+
+
+// List of names of possible strings to build a command out of
+enum CommandString_t {
+    VELOCITY,
+    CRUISE_ENABLE,
+    CRUISE_SET,
+    REGEN_ENABLE,
+    VALUE,
+    TEXT,
+    PCO,
+    SYSTEM,
+    PAGE,
+    ERROR0,
+    ERROR1,
+    ERROR2,
+    ERROR3,
+    ERROR4,
+    ERROR5
+};
+
+// The command strings themselves
+static char *CommandStrings[] = {
+    "x0",
+    "t1",
+    "t2",
+    "t3",
+    "val",
+    "txt",
+    "pco",
+    "",
+    "page",
+    "t4",
+    "t5",
+    "t6",
+    "t7",
+    "t8",
+    "t9"
+};
+
+/**
+ * Convert signed 32-bit integer to string
+ * Returns pointer to string
+ */
+static char *to_charp(int32_t num, char *dest) {
```

Comment:
If you want, you can do this with `sprintf(dest, "%ld", num)` instead of writing your own function